### PR TITLE
Generic wrapper

### PR DESCRIPTION
**Pull request by enoodle**
Generic function wrapper to use for any function.
This is aimed to be a base line for use with containers and for other wrappers.

### CODE REVIEW

File: example/generic_go_function/main.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	} else {
+		log.Printf("error was: %v", err)
+	}
+
+	// With Epsagon instrumentation
+	config := epsagon.NewTracerConfig("generic-go-wrapper", "")
+	config.Debug = true
+	response := epsagon.GoWrapper(config, doTask)(5, "hello")
+	res2 := response[0].Int()
+	errInterface := response[1].Interface()
```
Comment (bbernays) [inline example/generic_go_function/main.go:28]: Coming from python, I have no issue with this

File: example/generic_go_function/main.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+		log.Printf("First result is %d", res)
+	} else {
+		log.Printf("error was: %v", err)
+	}
+
+	// With Epsagon instrumentation
+	config := epsagon.NewTracerConfig("generic-go-wrapper", "")
+	config.Debug = true
+	response := epsagon.GoWrapper(config, doTask)(5, "hello")
+	res2 := response[0].Int()
```
Comment (adavidai) [inline example/generic_go_function/main.go:27]: Is it equivalent to `res, err = epsagon.GoWrapper(config, doTask)(5, "hello")`, then 
`res2 := res.Int()` and `errInterface := err.Interface()` ? 
(see https://gobyexample.com/errors)

Comment (enoodle) [reply example/generic_go_function/main.go:27]: It is equivalent in the meaning, but can't be written like that.

**Comment by ranrib** [comment]
Hi @enoodle please make sure to update the docs with:
https://github.com/epsagon/epsagon-go#integrations (the aws-sdk-go-v2 support)
and https://github.com/epsagon/epsagon-go#frameworks as a generic support (and please provide an example).

Thanks!

**Comment by enoodle** [comment]
@ranrib aws-sdk-go-v2 support is already there. ;-)

File: epsagon/generic_wrapper.go [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+				Traceback: string(debug.Stack()),
+				Time:      tracer.GetTimestamp(),
+			}
+			if wrapper.invoking {
+				wrapper.runner.Exception = exception
+				wrapper.runner.ErrorCode = protocol.ErrorCode_EXCEPTION
+				panic(userError{
+					exception: wrapper.thrownError,
+					stack:     exception.Traceback,
+				})
```
Comment (maorlx) [inline epsagon/generic_wrapper.go:95]: should wrapper.addRunnerEvent be called in that case?

File: epsagon/lambda_trigger.go [hunk 4]
```diff
@@ -422,6 +421,7 @@ func addLambdaTrigger(
 	payload json.RawMessage,
 	metadataOnly bool,
 	triggerFactories map[string]factoryAndType,
+	currentTracer tracer.Tracer,
```
Comment (maorlx) [inline epsagon/lambda_trigger.go:424]: is it used? (cant find param usage)

File: epsagon/generic_wrapper.go [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+	wrapper.addRunnerEvent()
+	return results
+}
+
+// GenericFunction type
+type GenericFunction func(args ...interface{}) []reflect.Value
+
+// GoWrapper wraps the function with epsagon's tracer
+func GoWrapper(config *Config, wrappedFunction interface{}) GenericFunction {
+	return func(args ...interface{}) []reflect.Value {
```
Comment (maorlx) [inline epsagon/generic_wrapper.go:120]: nt sure about the trace/events structure, but if a GoWrapper func is called from a lambda func then should it be a new tracer (like its being generated now) or used the lambda tracer?

File: epsagon/generic_wrapper.go (hunk 3 above)
Comment (enoodle) [reply epsagon/generic_wrapper.go:95]: :+1:

File: epsagon/generic_wrapper.go (hunk 5 above)
Comment (enoodle) [reply epsagon/generic_wrapper.go:120]: You shouldn't do it.

File: epsagon/lambda_trigger.go (hunk 4 above)
Comment (enoodle) [reply epsagon/lambda_trigger.go:424]: :+1:

**Pull request merged**